EIPS/eip-5320.md

```diff
@@ -0,0 +1,156 @@
+  /// fund was updated.
+  /// @param _tokenId The identifier of the token that had a fund change.
+  /// @param _amount The updated amount of the fund.
+  event TokenFunded(uint256 indexed _tokenId, uint256 _amount);
+
+  /// @dev emits when the valuation of an token changes.
+  /// @param _tokenId The identifier of the token whose valuation changed.
+  /// @param _valuation The new valuation of the token.
+  event ValuationSet(uint256 indexed _tokenId, uint256 _valuation);
+  
+  /// @dev emits when a token has tax paid for them.
+  /// @param _tokenId The identifier of the token that was taxed.
+  /// @param _value The taxes that have been paid.
+  event TaxPaid(uint256 indexed _tokenId, uint256 _value);
+
+  /// @dev buy a token. This function must succeed if 
+  /// @param _tokenId The identifier of the token to buy.
+  /// @param _offer The maximum amount offered to buy this token.
+  /// @param _valuation The valuation the token will have after a successful purchase.
+  /// @param _fund The fund put in to pay for the taxes of the token.
+  function buy(uint256 _tokenId, uint256 _offer, uint256 _valuation, uint256 _fund) external;
+
+  /// @dev call this to fund the item, to pay for the maintenance taxes.
+  /// @param _tokenId The identifier of the token to fund.
+  /// @param _value The amount sent to fund the token.
+  function fund(uint256 _tokenId, uint256 _value) external;
+
+  /// @dev call this to remove excess funding for the token.
+  /// if the resultant fund for the token was zero, it must be revoked.
+  /// @param _tokenId The identifier of the token to defund.
+  /// @param _value The amount to retrieve from the token's fund.
+  function defund(uint256 _tokenId, uint256 _value) external;
+  
+  /// @dev call this to revoke ownership of a token.
+  /// @param _tokenId The identifier of the token to revoke.
+  function revoke(uint256 _tokenId) external;
+
+  /// @dev call this to manually send pending taxes to the collector.
+  /// This function should be callable by any external party.
+  /// @param _tokenId The identifier of the token to pay taxes for.
+  function collect(uint256 _tokenId) external;
+  
+  /// @dev call this to change the perceived valuation of the token.
+  /// @param _tokenId The identifier of the token to change valuation of.
+  /// @param _valuation The new valuation of the token.
+  function changeValuation(uint256 _tokenId, uint256 _valuation) external;
+}
+```
+
+### Caveats
+
+The taxes have to be collected manually, either through `collect`, or by interacting with the Harberger tokens.
+
+The taxes are paid upfront, which is less capital efficient.
+
+The functions of the standard are not passing the buyer or owner as parameter, instead, they use `msg.sender`.
+
+The funds are per token, not per account.
+
+## Rationale
+
+### EIP-20 compatibility
+
+EIP-20 compliant tokens are a widespread industry standard and using native tokens is limiting.  
```

Comment:
That's for the Harberger Tax contract to decide. The deployer can define what the allowed currency is. For example, in [this implementation](https://github.com/kleros-crime-syndicate/harbs/blob/master/contracts/src/HarbergerAdsFull.sol) the currency is defined at construction time, so you can, for example, make the allowed currency be WETH.

Having a single, enforced currency is not enforced at the standard level, but ideally a contract shouldn't allow *any* EIP-20 token because it opens up these attacks.

Reply:
Could you put this in the Security Considerations?

Reply:
HT may be philosophically incompatible with fungibility because each token, by nature of having a self-assessed valuation, is non-fungible.